src/ncp/thread_host.hpp

```diff
@@ -170,6 +170,14 @@ class ThreadHost : virtual public NetworkProperties
 
 protected:
     static otLogLevel ConvertToOtLogLevel(otbrLogLevel aLevel);
+
+private:
+    static std::unique_ptr<ThreadHost> CreateThreadHost(CoprocessorType                  aType,
```

Comment:
Where should a vendor implement this method?

I feel it kind of inconspicuous that a vendor needs to implement this private static method.

Reply:
I've refactor it. Let's just use the static `Create` method for vendor to provide a different implementation.

In terms of Android, I plan to put the vendor implementation at `src/ncp/android`, which is a new directory. See this draft [CL](https://android-review.git.corp.google.com/c/platform/external/ot-br-posix/+/3305636).